plugins/nf-tower/src/main/io/seqera/tower/plugin/cli/AuthCommandImpl.groovy

```diff
@@ -0,0 +1,1354 @@
+            ColorUtil.printColored("${ColorUtil.colorize('nextflow auth login', 'cyan')} sets credentials using Nextflow config files, which take precedence over the environment variable.", "dim")
+            ColorUtil.printColored(" however, caution is advised to avoid confusing behaviour.", "dim")
+            println ""
+        }
+
+        // Check if seqera_auth.config file already exists
+        final authFile = getAuthFile()
+        if( Files.exists(authFile) ) {
+            ColorUtil.printColored("Error: Authentication token is already configured in Nextflow config.", "red")
+            ColorUtil.printColored("Auth file: ${ColorUtil.colorize(authFile.toString(), 'magenta')}", "dim")
+            println " Run ${ColorUtil.colorize('nextflow auth logout', 'cyan')} to remove the current authentication."
+            return
+        }
+
+        ColorUtil.printColored("Nextflow authentication with Seqera Platform", "cyan bold")
+        ColorUtil.printColored(" - Authentication will be saved to: ${ColorUtil.colorize(getAuthFile().toString(), 'magenta')}", "dim")
+
+        apiUrl = normalizeApiUrl(apiUrl)
+        ColorUtil.printColored(" - Seqera Platform API endpoint: ${ColorUtil.colorize(apiUrl, 'magenta')} (can be customised with ${ColorUtil.colorize('-url', 'cyan')})", "dim")
+
+        // Check if this is a cloud endpoint or enterprise
+        final endpointInfo = getCloudEndpointInfo(apiUrl)
+        if( endpointInfo.isCloud ) {
+            try {
+                performAuth0Login(endpointInfo.endpoint as String, endpointInfo.auth as Map)
+            } catch( Exception e ) {
+                log.debug("Authentication failed", e)
+                throw new AbortOperationException("${e.message}")
+            }
+        } else {
+            // Enterprise endpoint - use PAT authentication
+            handleEnterpriseAuth(apiUrl)
+        }
+    }
+
+    protected void performAuth0Login(String apiUrl, Map auth0Config) {
+
+        // Start device authorization flow
+        final deviceAuth = requestDeviceAuthorization(auth0Config)
+
+        println ""
+        ColorUtil.printColored("Confirmation code: ${ColorUtil.colorize(deviceAuth.user_code as String, 'yellow')}", "cyan bold")
+        final urlWithCode = "${deviceAuth.verification_uri}?user_code=${deviceAuth.user_code}"
+        println "${ColorUtil.colorize('Authentication URL:', 'cyan bold')} ${ColorUtil.colorize(urlWithCode, 'magenta')}"
+        ColorUtil.printColored("\n[ Press Enter to open in browser ]", "cyan bold")
+
+        // Wait for Enter key with proper interrupt handling
+        try {
+            final input = System.in.read()
+            if( input == -1 ) {
+                throw new AbortOperationException("Authentication cancelled")
+            }
+        } catch( InterruptedException e ) {
+            Thread.currentThread().interrupt()
+            throw new AbortOperationException("Authentication cancelled")
```

Comment:
This should not be needed, the ` Thread.currentThread().interrupt()` cause the blocking thread to throw an exception

Reply:
Removed in bb0ebe868